bin/node/src/executor.rs

```diff
@@ -1,27 +1,64 @@
-//! `CodeExecutor` specialization which uses natively compiled runtime when the WASM to be
-//! executed is equivalent to the natively compiled code.
+//! This module declares an `AlephExecutor` which is either a
+//! * `WasmExecutor`, for production and test build (when no local debugging is required)
+//! * `NativeElseWasmExecutor` for `try-runtime`, `runtime-benchmarks` and local debugging builds
 
-use sc_executor::NativeElseWasmExecutor;
+use sc_service::Configuration;
 
-// Declare an instance of the native executor named `ExecutorDispatch`. Include the wasm binary as the equivalent wasm code.
-pub struct ExecutorDispatch;
+#[cfg(not(any(
+    feature = "runtime-benchmarks",
+    feature = "local-debugging",
+    feature = "try-runtime"
+)))]
+pub mod aleph_executor {
+    use sc_executor::WasmExecutor;
+
+    use super::Configuration;
 
-impl sc_executor::NativeExecutionDispatch for ExecutorDispatch {
-    #[cfg(feature = "runtime-benchmarks")]
     type ExtendHostFunctions = (
+        sp_io::SubstrateHostFunctions,
         aleph_runtime_interfaces::snark_verifier::HostFunctions,
-        frame_benchmarking::benchmarking::HostFunctions,
     );
-    #[cfg(not(feature = "runtime-benchmarks"))]
-    type ExtendHostFunctions = (aleph_runtime_interfaces::snark_verifier::HostFunctions,);
+    pub type Executor = WasmExecutor<ExtendHostFunctions>;
+
+    pub fn get_executor(config: &Configuration) -> Executor {
+        sc_service::new_wasm_executor(config)
+    }
+}
+
+#[cfg(any(
+    feature = "runtime-benchmarks",
+    feature = "local-debugging",
+    feature = "try-runtime"
+))]
+pub mod aleph_executor {
+    use sc_executor::NativeElseWasmExecutor;
+
+    use super::Configuration;
 
-    fn dispatch(method: &str, data: &[u8]) -> Option<Vec<u8>> {
-        aleph_runtime::api::dispatch(method, data)
+    pub struct ExecutorDispatch;
+
+    impl sc_executor::NativeExecutionDispatch for ExecutorDispatch {
+        #[cfg(feature = "runtime-benchmarks")]
```

Comment:
Just to make sure - we don't need  `sp_io::SubstrateHostFunctions` for `NativeElseWasmExecutor`, right?

Reply:
Correct, `NativeElseWasmExecutor` constructor [takes care of it](https://github.com/paritytech/polkadot-sdk/blob/release-polkadot-v1.4.0/substrate/client/executor/src/executor.rs#L610).